src/modules/fundingManager/bondingCurve/FM_BC_Tools.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.23;
+
+library FM_BC_Tools {
+    //error Module__BondingCurveBase__FeeAmountToHigh();
+
+    /// @dev Converts an amount to a required decimal representation.
+    /// This function is useful for handling tokens with different decimal places.
+    /// It takes care of both upscaling and downscaling the decimals based on the required decimals.
+    ///
+    /// @param _amount The amount to be converted.
+    /// @param _tokenDecimals The current decimal places of the token.
+    /// @param _requiredDecimals The required decimal places for the token.
+    ///
+    /// @return The converted amount with required decimal places.
+    function _convertAmountToRequiredDecimal(
+        uint _amount,
+        uint8 _tokenDecimals,
+        uint8 _requiredDecimals
+    ) internal pure returns (uint) {
+        // If the token decimal is the same as required decimal, return amount
+        if (_tokenDecimals == _requiredDecimals) {
+            return _amount;
+        }
+        // If the decimal of token is > required decimal, calculate conversion rate and
+        // return amount converted to required decimal
+        if (_tokenDecimals > _requiredDecimals) {
+            uint conversionFactor = (10 ** (_tokenDecimals - _requiredDecimals));
+            return (_amount / conversionFactor);
+        } else {
+            // If the decimal of token is < required decimal, calculate conversion rate and
+            // return amount converted to required decimals
+            uint conversionFactor = (10 ** (_requiredDecimals - _tokenDecimals));
+            return (_amount * conversionFactor);
+        }
+    }
+
+    /*
```

Comment:
Outcommented

Reply:
Yeah, this is the one function were I think having it here SHOULD be more efficient, but it actually isn't when I redirect the calls. I wanted to wait to see if anybody has insights on this, if not I'll just remove it before merging.